src/app/initialisation/syncPoolService.js

```diff
@@ -43,14 +48,22 @@ function SyncPoolService($q, mediator, workorderSync, workflowSync, resultSync,
     promises.push(workflowSync.createManager());
     promises.push(messageSync.createManager());
     promises.push(resultSync.createManager());
-    return $q.all(promises).then(function(managers) {
-      var map = {};
-      managers.forEach(function(managerWrapper) {
-        map[managerWrapper.manager.datasetId] = managerWrapper;
+
+    //Initialising the sync managers for the required datasets.
+    return syncService.manage(config.datasetIds.workorders, {}, {}, config.syncOptions)
+      .then(syncService.manage(config.datasetIds.workflows, {}, {}, config.syncOptions))
+      .then(syncService.manage(config.datasetIds.results, {}, {}, config.syncOptions))
+      .then(syncService.manage(config.datasetIds.messages, {}, {}, config.syncOptions))
+      .then(function() {
+        return $q.all(promises).then(function(managers) {
+          var map = {};
+          managers.forEach(function(managerWrapper) {
+            map[managerWrapper.manager.datasetId] = managerWrapper;
+          });
+          map.workorders.manager.publishRecordDeltaReceived(mediator);
```

Comment:
[Question] Why this is happening in this module? Is that some optional behavior?